server/src/main/java/com/flipkart/varadhi/qos/TopicRateLimiter.java

```diff
@@ -0,0 +1,41 @@
+package com.flipkart.varadhi.qos;
+
+import com.google.common.util.concurrent.AtomicDouble;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class TopicRateLimiter implements RateLimiter {
+
+    @Getter
+    private final String topic;
+    @Getter
+    private final String name;
+    private volatile double suppressionFactor;
+    private AtomicDouble lastObserved;
+    private AtomicDouble currentObserved;
+
+    public TopicRateLimiter(String topic, String name) {
+        this.topic = topic;
+        this.name = name;
```

Comment:
what is name here?

Reply:
Renamed to type of rate limiter. Its enum now `RateLimiterType`.